cmd/kubeadm/app/constants/constants.go

```diff
@@ -208,8 +209,33 @@ var (
 
 	// MinimumKubeletVersion specifies the minimum version of kubelet which kubeadm supports
 	MinimumKubeletVersion = version.MustParseSemantic("v1.8.0")
+
+	// SupportedEtcdVersion lists officially supported etcd versions with corresponding kubernetes releases
+	SupportedEtcdVersion = map[string]string{
```

Comment:
To make this way easier, can you just use a `uint8` here that tracks the minor version of k8s?
i.e. 8 -> 3.0.17 and 9 -> 3.1.10

Reply:
the idea was to be flexible for the case when let's say 1.9 supports 3.1.10 but starting from 1.9.5 for example will start supporting 3.2.X or something. If you think that this use case is impossible, then I will switch to what you suggested.

Reply:
Yeah, I don't see that coming in any way. We haven't ever done something like that so far and I don't think we'll do that either. etcd upgrades are sensitive ;)

Reply:
done but instead of 8/9 I am using 18, 19 etc it is easier to map to a real k8s version.